js/ui/iconGrid.js

```diff
@@ -175,7 +213,7 @@ const BaseIcon = new Lang.Class({
             size = found ? len : ICON_SIZE;
         }
 
-        if (this.iconSize == size && this._iconBin.child)
+        if (this.iconSize == size && this.icon !== null)
```

Comment:
In eos-shell we used to check for a child. That doesn't work with `St.Widget` on `_layeredIcon`, so I'm just checking if `icon` is non-null here, which really checks the same invariant (has the icon been set yet?). I think on eos-shell this was actually broken since it would check`.child` which was always undefined and thus we would be re-creating the icons all the time.